### PR TITLE
chore: improve logs

### PR DESCRIPTION
**Pull request by AnguIar**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                      |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,8 +45,8 @@
     "@map-colonies/express-access-log-middleware": "^1.0.0",
     "@map-colonies/js-logger": "^0.0.5",
     "@map-colonies/mc-model-types": "^13.4.3",
```
Comment (asafMasa) [inline package.json]: update mc-model-types to 14.0.1

File: src/clients/jobManagerWrapper.ts [hunk 2]
```diff
@@ -181,23 +181,22 @@ export class JobManagerWrapper extends JobManagerClient {
...
-        this.logger.info({ jobId, oldExpirationDate, newExpirationDate, msg });
+        this.logger.info({ jobId, oldExpirationDate, newExpirationDate }, 'Will execute update for expirationDate');
         await this.put(getOrUpdateURL, {
           expirationDate: newExpirationDate,
         });
       } else {
-        const msg = 'New expiration date is older than current expiration date';
-        this.logger.info({ jobId, oldExpirationDate, newExpirationDate, msg });
+        const msg = 'Will not update expiration date, as current expiration date is later than current expiration date';
+        this.logger.debug({ jobId, oldExpirationDate, newExpirationDate, msg });
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts]: info

File: src/createPackage/models/createPackageManager.ts [hunk 3]
```diff
@@ -95,7 +95,7 @@ export class CreatePackageManager {
 
     const sanitizedBbox = this.sanitizeBbox(polygon as Polygon, layerMetadata.footprint as Polygon | MultiPolygon, zoomLevel);
     if (sanitizedBbox === null) {
-      throw new BadRequestError(`Requested bbox has no intersection with requested layer`);
+      throw new BadRequestError(`Requested ${polygon} has no intersection with requested layer ${layer.id}`);
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: fix lint

Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: did you made sure this is printed correctly and not { object }?

File: src/pollingManager.ts [hunk 4]
```diff
@@ -20,16 +20,16 @@ export class PollingManager {
     const jobs = await this.taskManager.getJobsByTaskStatus();
     const expirationDate = new Date();
     expirationDate.setDate(expirationDate.getDate() + this.expirationDays);
-    if ((jobs.completedJobs?.length ?? 0) > 0) {
+    if (jobs.completedJobs?.length) {
```
Comment (asafMasa) [inline src/pollingManager.ts:22]: fix lint

File: src/pollingManager.ts [hunk 5]
```diff
@@ -20,16 +20,16 @@ export class PollingManager {
...
       existsJobs = true;
-      this.logger.info(`completed jobs detected, running finalize job`);
-      for (const job of jobs.completedJobs as JobResponse[]) {
+      this.logger.info(`Completed jobs detected, running finalize job`);
+      for (const job of jobs.completedJobs) {
         await this.taskManager.finalizeJob(job, expirationDate);
       }
-    } else if ((jobs.failedJobs?.length ?? 0) > 0) {
+    } else if (jobs.failedJobs?.length) {
```
Comment (asafMasa) [inline src/pollingManager.ts:28]: fix lint

File: src/tasks/models/tasksManager.ts [hunk 6]
```diff
@@ -92,13 +92,13 @@ export class TasksManager {
       const promisesResponse = await Promise.allSettled(callbackPromises);
       promisesResponse.forEach((response, index) => {
         if (response.status === 'rejected') {
-          this.logger.error(`Did not send callback to ${targetCallbacks[index].url}, got error: ${JSON.stringify(response.reason)}`);
+          this.logger.error(response.reason, `Failed to send callback to ${targetCallbacks[index].url}`);
```
Comment (asafMasa) [inline src/tasks/models/tasksManager.ts]: shouldn't you put the reason and url as parameters? { url: ..., reason: ... }?

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/pollingManager.ts (hunk 4 above)
Comment (AnguIar) [reply src/pollingManager.ts:22]: It's ok. we don't want to handle nullish/zero/NaN cases explicitly

Comment (asafMasa) [reply src/pollingManager.ts:22]: but this can cause a bug as you removed the > 0 condition

File: src/pollingManager.ts (hunk 5 above)
Comment (asafMasa) [reply src/pollingManager.ts:28]: but this can cause a bug as you removed the > 0 condition

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/pollingManager.ts (hunk 4 above)
Comment (AnguIar) [reply src/pollingManager.ts:22]: `jobs.completedJobs.length` can either be a number (including 0) **OR** `undefined`.
If it's not a positive number (i.e `undefiend` or the value `0`) we should not enter the if statement.
And that's exactly what happens here

**Review by asafMasa — APPROVED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**